src/main/webapp/go.js

```diff
@@ -48,10 +65,89 @@ function getHunt() {
       puzzleArr.push(mssg.items[i].riddle.puzzle);
       destArr.push(mssg.items[i].name);
     }
+    createMap();
     updateToCurrentState(destIndex);
   });
 }
 
+/**
+ * Add a marker to the map at the specified location.
+ * @param {Double} destLat: latitude of location.
+ * @param {Double} destLng: longitude of location.
+ * @param {String} destName: Name of location.
+ */
+function addMarkerToMap(destLat, destLng, destName) {
+  const coord = new google.maps.LatLng(destLat, destLng);
+  const marker = new google.maps.Marker({
+    position: coord,
+    title: destName,
+  });
+  marker.setMap(map);
+}
+
+/**
+ * Creates a map and adds it to the page.
+ */
+function createMap() {
+  map = new google.maps.Map(
+      document.getElementById(MAP_DISPLAY),
+      // Centered at GooglePlex (updated below).
+      {center: {lat: 37.422, lng: -122.084}, zoom: 7},
+  );
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition((position) => {
+      const pos = {lat: position.coords.latitude,
+        lng: position.coords.longitude};
+      addMarkerToMap(position.coords.latitude, position.coords.longitude,
+          'Your current location');
+      map.setCenter(pos);
+    },
+    () => updateMessage(MAP_MSSG_DISPLAY,
+        'Error: I can\'t find your location.'));
+  } else {
+    updateMessage(MAP_MSSG_DISPLAY, 'Error: Your browser doesn\'t' +
+        'support geolocation.');
+  }
+}
+
+/**
+ * Show or hide the hint button.
+ * @param {boolean} hide Whether the proceed button should be hidden or shown.
+ * Disable lint check because this will later be called by
+ * handleDestinationAnswer(), updateToCurrentState(), and proceed().
+ *
+ */
+function toggleHintButton(hide) { //eslint-disable-line
+  const hintButton = document.getElementById(HINT_BUTTON);
+  if (hide) {
+    hintButton.classList.add(INVISIBLE_CLASS);
+  } else {
+    hintButton.classList.remove(INVISIBLE_CLASS);
+  }
+}
+
+/**
+ * Allow the user to see a hint.
+ * Disable lint check because getHint() is called from go.html.
+ */
+function getHint() { //eslint-disable-line
+  if (destIndex >= 0) {
+    const hintArr = huntArr[destIndex].hints;
+    if (hintIndex < hintArr.length) {
```

Comment:
I don't see where hintIndex is declared. I _think_ that `hintIndex` is being implicitly initialized upon its first mention. Could you explicitly declare `hintIndex` near the top of the file, à la destIndex?